pkg/urbit/worker/main.c

```diff
@@ -316,72 +324,80 @@ _serf_grab(u3_noun sac, u3_noun ovo, u3_noun vir)
   }
 }
 
-/* _serf_fail(): failure stub.
+/* _worker_fail(): failure stub.
 */
 static void
-_serf_fail(void* vod_p, const c3_c* wut_c)
+_worker_fail(void* vod_p, const c3_c* wut_c)
 {
-  fprintf(stderr, "serf: fail: %s\r\n", wut_c);
+  u3l_log("worker: fail: %s\r\n", wut_c);
   exit(1);
 }
 
-/* _serf_send(): send result back to lord.
+/* _worker_send(): send result back to daemon.
 */
 static void
-_serf_send(u3_noun job)
+_worker_send(u3_noun job)
 {
   u3_newt_write(&u3V.out_u, u3ke_jam(job), 0);
 }
 
-/* _serf_send_replace(): send replacement job back to lord.
+/* _worker_send_replace(): send replacement job back to daemon.
+*/
+static void
+_worker_send_replace(c3_d evt_d, u3_noun ovo)
+{
+  u3l_log("worker_send_replace %" PRIu64 " %s\r\n",
+          evt_d,
+          u3r_string(u3h(u3t(ovo))));
+
+  _worker_send(u3nq(c3__work,
+                    u3i_chubs(1, &evt_d),
+                    u3V.mug_l,
+                    u3nc(u3k(u3A->now), ovo)));
+}
+
+/* _worker_send_complete(): report completion.
 */
 static void
-_serf_send_replace(c3_d evt_d, u3_noun ovo)
+_worker_send_complete(u3_noun vir)
 {
-  fprintf(stderr, "serf_send_replace %" PRIu64 " %s\r\n",
-                  evt_d,
-                  u3r_string(u3h(u3t(ovo))));
-
-  _serf_send(u3nq(c3__work,
-                  u3i_chubs(1, &evt_d),
-                  u3V.mug_l,
-                  u3nc(u3k(u3A->now), ovo)));
+  _worker_send(u3nq(c3__done,
+                    u3i_chubs(1, &u3V.evt_d),
+                    u3r_mug(u3A->roc),
+                    vir));
 }
 
-/* _serf_send_complete(): report completion.
+/* _worker_send_stdr(): send stderr output
 */
 static void
-_serf_send_complete(u3_noun vir)
+_worker_send_stdr(c3_c* cstr)
 {
-  _serf_send(u3nq(c3__done,
-                  u3i_chubs(1, &u3V.evt_d),
-                  u3r_mug(u3A->roc),
-                  vir));
+  _worker_send(u3nt(c3__stdr, u3i_chubs(1, &u3V.evt_d), u3i_string(cstr)));
```

Comment:
I'd like to note that this could instead be `_worker_send_slog(u3nc(c3__leaf, u3i_tape(cstr_c)));`. Do you think it's important to separate the nock and runtime output streams?

Reply:
Philosophically, there feels like a difference between a `~&`'s output and the output of the interpreter itself. There's a difference between the runtime's output and the output of the program the runtime is running.

Reply:
Fine by me, just thought I should mention the option.